pkg/apply/desiredset_process_test.go

```diff
@@ -109,3 +113,157 @@ func Test_multiNamespaceList(t *testing.T) {
 		})
 	}
 }
+
+func Test_canIndexByHash(t *testing.T) {
```

Comment:
There is nothing to change here; it's just a suggestion for future tests. We would get a lot of mileage by having one test function for the entire `list(...)` function since we can better verify the desired interaction with the current behavior.